### PR TITLE
Pegasus: fix form editing

### PR DESCRIPTION
**Pull request by breville**
Attempting to edit some existing information about a volunteer at https://code.org/volunteer/engineer/edit/[secret] wasn't saving the changes.  This simple fix makes that work, by ensuring that before merging the new hash over the old, its keys are symbols so that they overwrite the appropriate fields.

### CODE REVIEW

**Comment by breville** [comment]
This will require a bunch of testing across all the forms.  I'm also not sure how this regressed.  I'm also curious about when `form_data` comes into effect [here](https://github.com/code-dot-org/code-dot-org/pull/24818/files#diff-5bd1e5c1e8d0292d16cb79aa7c247ad3L98).  What we are editing in this repro is what is usually encoded using JSON as `data`.

**Comment by joshlory** [comment]
Any way to add unit or integration test coverage for this?

**Comment by breville** [comment]
@joshlory thanks for asking.  There was a test that should have exercised this, and has now been updated to only work with the fix.

File: pegasus/test/test_form_helpers.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+require_relative '../src/env'
```
Comment (breville) [inline pegasus/test/test_form_helpers.rb:1]: This addition allows the test to be run standalone with 
```
bundle exec rake test TEST=test/test_form_helpers.rb
```

**Comment by breville** [comment]
I think this might have regressed with https://github.com/code-dot-org/code-dot-org/commit/39a5e9adc1be3bddffe07d0c065d0fd112b57eec in https://github.com/code-dot-org/code-dot-org/pull/17019.  Previously we were generating "symbolized" `data` (as well as symbolized `data[:data]`) but after this input data was renamed to `form`, it was no longer symbolized (though we were still symbolizing `form[:data]` as part of its JSON parsing).

The whole system is a bit confusing because we merge the contents of `form[:data]` and `form` before merging that result into the stored form object's `[:data]` entry.  (Not to mention taking a few values and setting them in the form object itself!)

**Comment by aoby** [comment]
Does this mean we might have a bunch of forms updated since the regression which have the updated value stored under a string key, alongside the original value with its symbol key? If so, do we have a plan to fix them?

**Comment by breville** [comment]
@aoby I don't believe so, because the [calls](https://github.com/code-dot-org/code-dot-org/pull/24818/files#diff-5bd1e5c1e8d0292d16cb79aa7c247ad3L100) to [validate_form](https://github.com/code-dot-org/code-dot-org/blob/47379314467b1bad950908600ef9473b893dd8e7/lib/forms/pegasus_form_validation.rb#L131) would not have allowed the string keys through.  (See the various `normalize` functions [under /pegasus/forms](https://github.com/code-dot-org/code-dot-org/search?q=normalize+path%3A%2Fpegasus%2Fforms&unscoped_q=normalize+path%3A%2Fpegasus%2Fforms).)

**Pull request merged**